logtail/logtail_test.go

```diff
@@ -23,29 +28,245 @@ func TestFastShutdown(t *testing.T) {
 	l := NewLogger(Config{
 		BaseURL: testServ.URL,
 	}, t.Logf)
-	l.Shutdown(ctx)
+	err := l.Shutdown(ctx)
+	if err != nil {
+		t.Error(err)
+	}
 }
 
-func TestUploadMessages(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
-	uploads := 0
-	testServ := httptest.NewServer(http.HandlerFunc(
+type LogtailTestServer struct {
+	ctx      context.Context
+	cancel   context.CancelFunc
+	uploaded chan int
+	logLines int // number of messages to be received
+	body     []byte
+	bodytext string
+	mu       sync.Mutex
+	srv      *httptest.Server
+}
+
+func NewLogtailTestServer(logLines int) *LogtailTestServer {
```

Comment:
I think if you accept a `t *testing.T` as an argument here, you can do `t.Cleanup(ts.srv.Close)`, which'll spare you having to call `Close` in each of the tests that use this server.

Reply:
Done.